packages/framework/src/Form/Admin/Transport/TransportFormType.php

```diff
@@ -129,10 +133,19 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             'label' => t('Prices'),
         ]);
 
-        $builderPricesGroup->add('pricesByDomains', PriceAndVatTableByDomainsType::class, [
-            'pricesIndexedByDomainId' => $this->transportFacade->getPricesIndexedByDomainId($transport),
-            'inherit_data' => true,
-            'render_form_row' => false,
+        $optionsByDomainId = [];
+
+        foreach ($this->domain->getAllIds() as $domainId) {
+            $optionsByDomainId[$domainId] = [
+                'domain_id' => $domainId,
+                'current_transport_prices_indexed_by_id' => $transport instanceof Transport ? $this->transportFacade->getPricesIndexedByTransportPriceId($transport) : [],
```

Comment:
No no, it returns the same array of all transports regardless of their domain. Basically in each of the multidomain form entries, I am interested in "half" of the array. I tried rewriting it in 88108de00b so each `TransportPricesWithWeightLimitsAndVatSelectType` gets only the prices on the relevant domain, however, I do not think it is somehow better, it is just another additional logic that actually does not improve anything. Maybe a better approach would be  calling `$this->transportFacade->getPricesIndexedByTransportPriceId($transport)` before the domain ids are iterated to avoid multiple calls, and just pass the same array to the all domain entries, what do you think?

Reply:
fixed in 7b586f2500